lib/dbs.js

```diff
@@ -49,11 +49,25 @@ const retryHandler = {
   }
 }
 
-function getLogsDb () {
+function getLogsDbName () {
   const date = new Date()
-  const currentLogDbName = 'logs-' + date.toISOString().substr(0, 7) + (env.isProduction ? '' : '-staging')
-  const db = new PouchDB(resolve(env.COUCH_URL, currentLogDbName))
-  return new Proxy(db, retryHandler)
+  return 'logs-' + date.toISOString().substr(0, 7) + (env.isProduction ? '' : '-staging')
+}
+
+let currentLogsDbName = getLogsDbName()
+let currentLogsDb
+
+function getLogsDb () {
+  const logsDbName = getLogsDbName()
+  if (!currentLogsDb || logsDbName !== currentLogsDbName) {
+    // either we are just starting up and there is no db
+    // instance yet, or we switched over in the month, so
+    // we need a new db instance
```

Comment:
will the old DB instance continue to exist forever?

Reply:
it should garbage collect after this last reference is overwritten.

Reply:
if not, we leak one instance per month, that’s a lot better than having ~1k+ instances on load spikes :D

Reply:
good question tho!